### PR TITLE
moved datastore to navmazing

### PR DESCRIPTION
**Pull request by izapolsk**
changed Datastore object to use navmazing library instead of older nav approach
{{pytest: cfme/tests/infrastructure/test_datastore_analysis.py cfme/tests/infrastructure/test_delete_infra_object.py cfme/tests/infrastructure/test_instance_analysis.py cfme/tests/cloud_infra_common/test_tag_objects.py cfme/tests/configure/test_default_views_infra.py cfme/tests/configure/test_visual_infra.py cfme/tests/infrastructure/test_set_default_filter.py}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 08f65da3e1862217e62fb170fa46997cdb88ab93:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/infrastructure/datastore.py [hunk 1]
```diff
@@ -85,28 +66,23 @@ def delete(self, cancel=True):
...
-        sel.force_navigate('infrastructure_datastores')
         wait_for(lambda: not self.exists, fail_condition=False,
              message="Wait datastore to disappear", num_sec=500, fail_func=sel.refresh)
 
     def wait_for_appear(self):
-        sel.force_navigate('infrastructure_datastores')
         wait_for(lambda: self.exists, fail_condition=False,
              message="Wait datastore to appear", num_sec=1000, fail_func=sel.refresh)
 
-    def load_details(self, refresh=False):
```
Comment (psav) [inline cfme/infrastructure/datastore.py]: Do we know if refresh is used anywhere in the source?

File: cfme/infrastructure/datastore.py [hunk 2]
```diff
@@ -85,28 +66,23 @@ def delete(self, cancel=True):
...
-        if not self._on_detail_page():
-            logger.debug("load_details: not on details already, navigating")
-            sel.force_navigate('infrastructure_datastore', context=self._get_context())
+    def load_details(self):
```
Comment (psav) [inline cfme/infrastructure/datastore.py]: I'm kinda thinking that we mandate the ```provider``` for a datastore, though I understand that makes it difficult if we try to *list* all providers. We would potentially need to read information from the db to find the provider. Perhaps this is a secondary action.

Comment (izapolsk) [inline cfme/infrastructure/datastore.py]: I agree with you that it is needed.
However I'd propose to do this in another PR for all related objects as well as I did for cluster.

**Comment by dajoRH** [comment]
Lint report for commit 9dbe350635d65708a89e125fb08bdc2c3ced06f2:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 96af1c66e4db9ea3662291d4ea5aba8f400e574b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/infrastructure/datastore.py (hunk 1 above)
Comment (psav) [inline cfme/infrastructure/datastore.py]: No, I'd like to remove the usage of load_details, and in particular the refresh behavior

File: cfme/infrastructure/datastore.py (hunk 2 above)
Comment (psav) [inline cfme/infrastructure/datastore.py]: Agreed

File: cfme/infrastructure/datastore.py (hunk 1 above)
Comment (izapolsk) [inline cfme/infrastructure/datastore.py]: As agreed, this change will be left as is and changed later. todo will be added.

**Comment by dajoRH** [comment]
Lint report for commit 9c3ee1752dfefd8b43c9c8808426bdcd67c08683:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9ce1114922cf7051e0c453282af4e579a168e222:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*
